File: back-end/src/main/java/com/okta/mongodb/GeneradoScripts/model/subscription/Subscription.java
package com.okta.mongodb.GeneradoScripts.model.subscription;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;

import com.okta.mongodb.GeneradoScripts.model.plan.Plan;
import com.okta.mongodb.GeneradoScripts.model.user.User;

@Entity
@Data
public class Subscription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "plan_id")
    private Plan plan;
    private Double price;
    private String method;

    private LocalDate startDate;
    private LocalDate endDate;
    private String status;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "transaccion_id")
    private Transaction transaction;
}

File: front-end/src/types/subscription/subscription.ts
import type { Plan } from "../plan/plan";
import type { User } from "../user/user";

export interface Subscription {
    id: number;
    endDate: string;
    method: string;
    price: number;
    startDate: string;
    status: string;
    plan: Plan;
    user: User;
}